dymension/libs/kaspa/lib/core/src/api/deposits.rs

```diff
@@ -82,33 +82,50 @@ impl HttpClient {
         Self { url, client: c }
     }
 
-    pub async fn get_deposits(&self, address: &str) -> Result<Vec<Deposit>> {
-        let limit = 20;
-        let lower_bound = Some(0i64);
-        let upper_bound = Some(0i64);
-        let field = None;
-        let resolve_previous_outpoints = None;
-        let acceptance = None;
+    pub async fn get_deposits_by_address(&self,start_time: i64,address: &str) -> Result<Vec<Deposit>> {
+        let limit: i64 = 100;
+        let mut upper_bound = 0i64;
+        let lower_bound = start_time;
 
         let c = self.get_config();
         info!("Dymension query kaspa deposits, url: {:?}", c.base_path);
 
-        let res = transactions_page(
-            &c,
-            args {
-                kaspa_address: address.to_string(),
-                limit: Some(limit),
-                before: lower_bound,
-                after: upper_bound,
-                fields: field,
-                resolve_previous_outpoints: resolve_previous_outpoints,
-                acceptance: acceptance,
-            },
-        )
-        .await?;
-
-        Ok(res
+        let mut txs: Vec<TxModel> = Vec::new();
+
+        loop {
+            let mut res = transactions_page(
+                &c,
+                args {
+                    kaspa_address: address.to_string(),
+                    limit: Some(limit),
+                    before: Some(upper_bound),
+                    after: Some(lower_bound),
+                    fields: None,
+                    resolve_previous_outpoints: None,
+                    acceptance: Some(AcceptanceMode::Accepted),
+                },
+            )
+            .await?;
+            txs.append(&mut res);
+            if res.len() < limit as usize {
+                break;
+            }
+            // txs should be in descendent order, so we save last returned tx time and we continue from there
+            if let Some(last_val) = res.last() {
+                if let Some(block_time) = last_val.block_time {
+                    upper_bound = block_time + 1;
+                }
+            }
```

Comment:
is there a concept of order when the chain can reorg though?
Every node has a different view of the order no? it's only reliable for mature data